api/src/main/java/org/apache/iceberg/types/TypeUtil.java

```diff
@@ -136,7 +136,16 @@ public static Types.StructType selectNot(Types.StructType struct, Set<Integer> f
   }
 
   public static Schema selectNot(Schema schema, Set<Integer> fieldIds) {
-    Set<Integer> projectedIds = getIdsInternal(schema.asStruct(), false);
+    return selectNot(schema, fieldIds, false);
+  }
+
+  public static Schema selectNotPreserveEmptyStructs(Schema schema, Set<Integer> fieldIds) {
```

Comment:
maybe we should just have a single utility method for reading the schema without constants and without metadata fields rather than doing it in multiple places? Then we wouldn't need to expose `selectNotPreserveEmptyStructs`. wdyt?

Reply:
That makes sense. Since ORC is the only format using this behavior, I now added this code to `OrcSchemaUtil`.